libcxx/test/std/algorithms/alg.nonmodifying/alg.fold/requirements.h

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
```

Comment:
Could you please take another pass at naming these types. They often name concepts they don't fulfill or fulfill only in one particular case. 

Consider moving some of them closer to their test.

Reply:
PTAL